### PR TITLE
{bio}[foss/2020a] TEtranscripts 2.2.0 w/ Python 2.7.18 & 3.8.2

### PR DESCRIPTION
**Pull request by hajgato**

### CODE REVIEW

**Comment by hajgato** [comment]
This is somewhat related to `TEToolkit-1.5.1-foss-2016a-Python-2.7.11.eb`, they look rather similar (`TEToolkit` has also `TEtranscripts`), but since we got a request for `TETranscripts` and both `TEToolkit` and `TEtranscripts` have different pypi pages, I decided to use a new name.

**Comment by hajgato** [comment]
I made it deliberately Python2, because I am not 100% sure that it is thoroughly tested with Python3. I have investigated the latest py2py3 PR of TEtranscripts, and it is nothing else than update of readme/documentation. 
There is one stronger argument though: we do have binaries, and as far as I know, multi_deps cannot handle multiple binaries. (so lets say TEtranscript loaded with Python3, the binaries would be still run with Python2 (or the latest installation iteration), as binaries from the latest installation, so the shebang is
`#!/apps/gent/CO7/skylake-ib/software/Python/2.7.18-GCCcore-9.3.0/bin/python`
This is not acceptable for me, not sure we do have any solution for this.

**Comment by lexming** [comment]
@hajgato the shebang in the scripts located inside `bin` can be fixed in the easyconfig with
```
fix_python_shebang_for = ['bin/*']
```
This will remove the absolute path to python

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node127.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/a6299a135c7acedcfb4ba3f92ccbb1e9 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node379.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/dbcfd95c3f923a7f38982e14b6f49616 for a full test report.

**Comment by lexming** [comment]
Going in, thanks @hajgato !

File: easybuild/easyconfigs/t/TEtranscripts/TEtranscripts-2.2.0-foss-2020a.eb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+download_dep_fail = True
+use_pip = True
+sanity_pip_check = True
+
+options = {'modulename': 'TEToolkit'}
+
+fix_python_shebang_for = ['bin/*']
+
+sanity_check_paths = {
+    'files': ['bin/TEtranscripts', 'bin/TEcount'],
```
Comment (boegel) [inline easybuild/easyconfigs/t/TEtranscripts/TEtranscripts-2.2.0-foss-2020a.eb:33]: @hajgato When binaries/scripts are involved, it's a good idea to include a sanity check command if possible (something like `TEcount --version` or `TEcount --help`) via `sanity_check_commands`, especially if `multi_deps` is involved.

Comment (boegel) [reply easybuild/easyconfigs/t/TEtranscripts/TEtranscripts-2.2.0-foss-2020a.eb:33]: fixed in #10921

**Pull request merged**